### PR TITLE
Standartize params for BeaconChain and Validator projects

### LINKED ISSUE #656: Standardize params.Config in Beacon-Chain and Validator Projects
Hi all,

We currently have two different ways of getting configuration params in both projects, the beacon chain uses params.GetConfig() while the validator uses params.DemoConfig() or params.DefaultConfig(). This is a bit confusing and would be best if we just stick to the way the beacon chain does so. This obfuscates what type of config parameters the system is using and lets the cli and node handle this effectively through a flag. This is preferred.

A PR to close this would change validator's params package to match the way the beacon-chain handles its params.

### PR DESCRIPTION
**Pull request by cyberbono3**
Resolves #656

Description
Standartize params in Beacon-Chain and Validator Projects

move validator/params and beacon-chain/params to shared/params
instead of SetEnv define SetDemoBeaconConfig() that handles config parameters
for clarity rename functions that handle validator and beacon-chain configs.

### CODE REVIEW

**Comment by prestonvanloon** [comment]
@terenc3t Please take another look

File: shared/params/config.go [hunk 1]
```diff
@@ -110,28 +111,65 @@ const (
 	Exit
 )
 
-// GetConfig retrieves beacon node config.
-func GetConfig() *Config {
+type OverrideConfig struct {
```
Comment (prestonvanloon) [inline shared/params/config.go]: This needs a godoc comment. Golint is complaining

File: shared/params/config.go [hunk 2]
```diff
@@ -110,28 +111,65 @@ const (
...
+	Param string
+}
+
+var emptyOverrideConfig = &OverrideConfig{}
+var demoOverrideConfig = &OverrideConfig{Param: "demo"}
+var customOverrideConfig = &OverrideConfig{Param: "custom"}
+
+func SetCustomConfig(c *Config) {
```
Comment (prestonvanloon) [inline shared/params/config.go]: This needs a godoc comment. Golint is complaining

File: shared/params/config.go [hunk 3]
```diff
@@ -110,28 +111,65 @@ const (
...
+	SetOverrideConfig(customOverrideConfig)
+	customConfig = c
+}
+
+func SetDemoBeaconConfig() {
```
Comment (prestonvanloon) [inline shared/params/config.go]: This needs a godoc comment. Golint is complaining

File: shared/params/config.go [hunk 4]
```diff
@@ -110,28 +111,65 @@ const (
...
+	SetOverrideConfig(demoOverrideConfig)
+}
+
+func SetOverrideConfig(oc *OverrideConfig) {
```
Comment (prestonvanloon) [inline shared/params/config.go]: This needs a godoc comment. Golint is complaining

File: shared/params/config.go [hunk 5]
```diff
@@ -110,28 +111,65 @@ const (
...
+	emptyOverrideConfig = oc
+}
+
+func GetOverrideConfig() *OverrideConfig {
```
Comment (prestonvanloon) [inline shared/params/config.go]: This needs a godoc comment. Golint is complaining

File: shared/params/config.go [hunk 6]
```diff
@@ -1,7 +1,7 @@
-// Package params defines important constants that are essential to the beacon chain.
```
Comment (terenc3t) [inline shared/params/config.go]: Please add a package comment there

File: beacon-chain/types/block.go [hunk 7]
```diff
@@ -192,10 +192,7 @@ func (b *Block) IsValid(
 
 	cStateProposerRandaoSeed := cState.Validators()[proposerIndex].RandaoCommitment
 	blockRandaoReveal := b.RandaoReveal()
-
-	// If this is a block created by the simulator service (while in development mode), we
```
Comment (nisdas) [inline beacon-chain/types/block.go]: why was this comment deleted ?

**Comment by prestonvanloon** [comment]
I'll take over this PR so we can merge it today

**Comment by prestonvanloon** [comment]
Notable changes:
- `GetBeaconConfig` -> `BeaconConfig`
- 'SetDemoBeaconConfig` -> `UseDemoBeaconConfig`
- Added godoc comments where necessary

**Pull request merged**